layers.go

```diff
@@ -72,6 +73,28 @@ type Layer struct {
 	// is set before using it.
 	Created time.Time `json:"created,omitempty"`
 
+	// CompressedDigest is the digest of the blob that was last passed to
+	// ApplyDiff() or Put(), as it was presented to us.
+	CompressedDigest digest.Digest `json:"compressed-diff-digest,omitempty"`
```

Comment:
Deduplication would be possible for the overlay driver, which stores layers as diffs, but not for the devicemapper driver, which stores layers as the filesystems that are produced by applying multiple layers on top of each other.  I think deduplication would have to be done on, or at least determined on, a per-driver basis.